frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx

```diff
@@ -2,57 +2,133 @@ import { useCallback } from "react";
 import { t } from "ttag";
 import _ from "underscore";
 
-import type { CollectionId, TableId } from "metabase-types/api";
+import { getQuestionVirtualTableId } from "metabase-lib/v1/metadata/utils/saved-questions";
+import type { CollectionItemModel, TableId } from "metabase-types/api";
 
-import type { EntityPickerModalOptions, EntityTab } from "../../EntityPicker";
+import type { EntityTab } from "../../EntityPicker";
 import { EntityPickerModal, defaultOptions } from "../../EntityPicker";
-import type { NotebookDataPickerValueItem, TablePickerValue } from "../types";
+import type { QuestionPickerItem } from "../../QuestionPicker";
+import { QuestionPicker } from "../../QuestionPicker";
+import { useHasModels, useHasQuestions } from "../hooks";
+import type {
+  DataPickerModalOptions,
+  DataPickerValue,
+  NotebookDataPickerValueItem,
+} from "../types";
+import {
+  isModelItem,
+  isQuestionItem,
+  isTableItem,
+  isValidValueItem,
+} from "../utils";
 
 import { TablePicker } from "./TablePicker";
 
 interface Props {
-  /**
-   * TODO: use this prop in https://github.com/metabase/metabase/issues/40719
-   */
-  collectionId: CollectionId | null | undefined;
```

Comment:
`collectionId` turned out to be leftover dead code.
It was introduced in #19009 but it became obsolete in #30558.